runelite-client/src/main/java/net/runelite/client/plugins/deathindicator/DeathIndicatorPlugin.java

```diff
@@ -0,0 +1,311 @@
+import net.runelite.api.events.ChatMessage;
+import net.runelite.api.events.ConfigChanged;
+import net.runelite.api.events.GameStateChanged;
+import net.runelite.api.events.GameTick;
+import net.runelite.api.events.LocalPlayerDeath;
+import net.runelite.client.config.ConfigManager;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.ui.overlay.infobox.InfoBoxManager;
+import net.runelite.client.ui.overlay.infobox.Timer;
+import net.runelite.client.ui.overlay.worldmap.WorldMapPointManager;
+
+@PluginDescriptor(
+	name = "Death Indicator"
+)
+@Slf4j
+public class DeathIndicatorPlugin extends Plugin
+{
+	public static final BufferedImage WORLDMAP_ARROW;
+	public static final BufferedImage BONES;
+
+	private Timer deathTimer;
+
+	@Inject
+	private Client client;
+
+	@Inject
+	private DeathIndicatorConfig config;
+
+	@Inject
+	private WorldMapPointManager worldMapPointManager;
+
+	@Inject
+	private InfoBoxManager infoBoxManager;
+
+	@Inject
+	private ConfigManager configManager;
+
+	static
+	{
+		try
+		{
+			synchronized (ImageIO.class)
+			{
+				WORLDMAP_ARROW = ImageIO.read(DeathIndicatorPlugin.class.getResourceAsStream("worldmap_arrow.png"));
+				BONES = ImageIO.read(DeathIndicatorPlugin.class.getResourceAsStream("bones.png"));
+			}
+		}
+		catch (IOException e)
+		{
+			throw new RuntimeException(e);
+		}
+	}
+
+	@Provides
+	DeathIndicatorConfig deathIndicatorConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(DeathIndicatorConfig.class);
+	}
+
+	@Override
+	protected void shutDown() throws Exception
+	{
+		if (client.hasHintArrow())
+		{
+			client.clearHintArrow();
+		}
+
+		if (deathTimer != null)
+		{
+			infoBoxManager.removeInfoBox(deathTimer);
+			deathTimer = null;
+		}
+
+		worldMapPointManager.removeIf(DeathWorldMapPoint.class::isInstance);
+	}
+
+	@Subscribe
+	public void onLocalPlayerDeath(LocalPlayerDeath event)
+	{
+		if (client.isInInstancedRegion())
+		{
+			return;
+		}
+
+		config.hasDied(true);
+
+		config.hasRespawned(false);
+
+		config.deathLocationX(client.getLocalPlayer().getWorldLocation().getX());
+		config.deathLocationY(client.getLocalPlayer().getWorldLocation().getY());
+		config.deathLocationPlane(client.getLocalPlayer().getWorldLocation().getPlane());
+
+		config.deathWorld(client.getWorld());
+
+		if (config.showDeathHintArrow())
+		{
+			client.setHintArrow(new WorldPoint(config.deathLocationX(), config.deathLocationY(), config.deathLocationPlane()));
+		}
+
+		if (config.showDeathOnWorldMap())
+		{
+			worldMapPointManager.removeIf(DeathWorldMapPoint.class::isInstance);
+			worldMapPointManager.add(new DeathWorldMapPoint(new WorldPoint(config.deathLocationX(), config.deathLocationY(), config.deathLocationPlane())));
+		}
+
+		if (deathTimer != null)
+		{
+			infoBoxManager.removeInfoBox(deathTimer);
+			deathTimer = null;
+		}
+
+		if (config.showDeathInfoBox())
+		{
+			config.timeOfDeath(Instant.now());
```

Comment:
I would always save this, in case the user toggles on the info box you would need this data.